go/v1beta1/storage/esutil/client.go

```diff
@@ -0,0 +1,357 @@
+// Copyright 2021 The Rode Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package esutil
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/elastic/go-elasticsearch/v7/esapi"
+	"github.com/rode/grafeas-elasticsearch/go/v1beta1/storage/filtering"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+type CreateRequest struct {
+	Index   string
+	Refresh string // TODO: use RefreshOption type
+}
+
+type GetRequest struct {
+	Index  string
+	Search *EsSearch
+}
+
+type ListRequest struct {
+	Index       string
+	Filter      string
+	Query       *EsSearch
+	Pagination  *ListPaginationOptions
+	SortOptions *ListSortOptions
+}
+
+type ListPaginationOptions struct {
+	Size      int
+	Token     string
+	Keepalive string
+}
+
+type ListSortOptions struct {
+	Direction EsSortOrder
+	Field     string
+}
+
+type ListResponse struct {
+	Hits          *EsSearchResponseHits
+	NextPageToken string
+}
+
+type UpdateRequest struct {
+	Index      string
+	DocumentId string
+	Refresh    string // TODO: use RefreshOption type
+}
+
+type DeleteRequest struct {
+	Index   string
+	Search  *EsSearch
+	Refresh string // TODO: use RefreshOption type
+}
+
+const defaultPitKeepAlive = "5m"
+const grafeasMaxPageSize = 1000
+
+type Client interface {
```

Comment:
Might be nice to have the `bulk` call here too, especially with the `ndjson` encoding.

Reply:
And if we want to replace using the `go-elasticsearch` client directly, we'll need msearch and mget exposed as well.